src/ddspdrum/module.py

```diff
@@ -166,20 +165,47 @@ def __init__(
         )
 
     def __call__(self, sustain_duration: float = 0):
-        """
-        Generate an ADSR envelope.
+        """Generate an ADSR envelope.
+
+        By default, this envelope reacts as if it was triggered with midi, for
+        example playing a keyboard. Each midi event has a beginning and end:
+        note-on, when you press the key down; and note-off, when you release the
+        key. `sustain_duration` is the amount of time that the key is depressed.
+
+        During the note-on, the envelope moves through the attack and decay
+        sections of the envelope. This leads to musically-intuitive, but
+        programatically-counterintuitive behaviour:
+
+        Assume attack is 0.5 seconds, and decay is 0.5 seconds. If a note is
+        held for 0.75 seconds, the envelope won't traverse through the entire
+        attack-and-decay (specifically, it will execute the entire attack, and
+        0.25 seconds of the decay).
```

Comment:
Understood. I'm slightly leaning towards, based upon discussion @jorshi was in, that we just assert if the params are not programmatically sensible. i.e. assert attack + decay <= sustain_duration

Reply:
It's often sensible to hold a note-on for less than A + D times. Happens all the time when you're playing an instrument with a fixed envelope (ie. 99.9999% of the case)

Reply:
okay, sounds good